### DISCUSSION TITLE
Exception thrown when running `route:list --sort`

### DISCUSSION THREAD

**Discussion by vanetix** [laravel/framework#38992] [Q&A]
- Laravel Version: `8.56.0`
- PHP Version: `7.4.23`
- Database Driver & Version: N/A

### Description:

Currently the help text for `route:list` shows that without an option value, `uri` will be used. This currently throws an exception since sort is undefined [here](https://github.com/laravel/framework/blob/3d687d93b4fe4232ecfd13efae29aa540c5b17c7/src/Illuminate/Foundation/Console/RouteListCommand.php#L130-L135) without passing the option.

<details>

<summary>Stacktrace</summary>

```bash
./artisan route:list --sort

   ErrorException

  Undefined index:

  at vendor/laravel/framework/src/Illuminate/Foundation/Console/RouteListCommand.php:133
    129▕      */
    130▕     protected function sortRoutes($sort, array $routes)
    131▕     {
    132▕         return Arr::sort($routes, function ($route) use ($sort) {
  ➜ 133▕             return $route[$sort];
    134▕         });
    135▕     }
    136▕
    137▕     /**

      +19 vendor frames
  20  artisan:37
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
```

</details>

### Steps To Reproduce:

Execute `route:list` without a sort value: `./artisan route:list --sort`. The expectation is the route list is output, sorting by `uri`.

I can throw a pull request together if the default `uri` sort is still the intended functionality.

**Comment by Krisell** [comment]
I think what the help text intends to say is that the default sorting order is by `uri` when the sorting option is not used at all.

However, I agree that using `uri` as the default even with the option makes sense, since leaving out the parameter for `--name` or `--path` doesn't throw an exception. I suggest you make a PR for this.

**Comment by crynobone** [comment]
Hey there,

We're closing this issue because it's inactive, already solved, old, or not relevant anymore. Feel free to open up a new issue if you're still experiencing this problem.
